pkg/gofr/http/errors.go

```diff
@@ -2,11 +2,18 @@
 package http
 
 import (
+	"errors"
 	"fmt"
 	"net/http"
 	"strings"
 )
 
+var errCommon = errors.New("HTTP error")
+
+func IsHTTPError(err error) bool {
+	return errors.Is(err, errCommon)
+}
```

Comment:
I think this might be OK for now, but a better name and sentinel could be provided.

I  say that because I didn't add the errCommon to ErrPanicRecovery, yet.

I could add another sentinel then, but also a new methods.

I struggle to find a name for the "is non critical error" and the sentinel.

Also, I dislike the obesity it brings.

Another solution could be to export the sentinels. But it's a different direction and it might be more difficult to maintain, also the magic behind and advantage of the Unwrap could be less visible.

Another way could be to provide another way to catch this, and only rely on the interface of having a status code and having a logic of 400 < code < 499

Anyway, if I keep IsHTTPError I will have to include PanicRecovery maybe

Here is the status of a Work in progress in my mind, what do you think guys

Reply:
@ccoVeille  I think for now we can keep `IsHTTPError` as it is simplifying the things and at the same time serving the logic we want to implement.
You can include a different error for panicRecovery as it will need to be logged on Error level only.

Reply:
I would use something like "IsRegularError()"

Reply:
I'm working on it again

Reply:
I implemented here what I initially planned